install.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+
+# This script is meant to automate the deployment of Szurubooru via Docker/Docker-Compose,
+# and to automatically take the necessary steps to allow Szurubooru to run on ARM devices
+# such as the Raspberry Pi.
+
+###########################################[Notes]###########################################
+# * Written and tested on Raspberry Pi 4 running Raspbian 10 (armv7), heavily based on a fork
+#   by kaijuagenda for the Pi 3: https://github.com/kaijuagenda/szurubooru-rpi3
+#
+# * Modifies docker-compose.yml rather than relying on a static alternative file to hopefully
+#   avoid the need for parallel maintenance of two files.
+#############################################################################################
+
+# Check for root to ensure the accessibility of the Docker daemon, and to 
+# allow mount point permissions to be set after building.
+
+if [[ $EUID -ne 0 ]]; then
+    echo "Sorry, but this script needs root to properly function.  Please try again using sudo ./install.sh" 1>&2
+    exit 1
+fi
+
+echo "Welcome to Szurubooru.  This install script will ask you to set a few 
+echo "config values, then get szurubooru up and running."
+
+# Check to ensure Docker and Docker-Compose are both installed before proceeding
+if ! command -v docker &> /dev/null || ! command -v docker-compose &> /dev/null; then
+    echo "Both Docker and Docker-Compose must already be installed to use this script."
+    echo "Please ensure both are installed and accessible using commands 'docker' and 'docker-compose', then run this script again."
+    exit 1
+else echo "Docker and Docker-Compose verified.  Continuing installation..."
+fi
+
+########################################################################################
+# Check CPU architectue and, if ARM (such as Raspberry Pi), modify docker-compose.yml
+# https://github.com/rr-/szurubooru/wiki/ARM-and-Raspberry-Pi-Support-for-Docker-builds
+# https://github.com/rr-/szurubooru/blob/master/docker-compose.yml
+########################################################################################
+
+if [[ "$(uname -m)" == 'a'* ]]; then
+    echo "ARM architecture detected.  Modfying docker-compose.yml accordingly."
+    sed -zi "s|image: szurubooru/server:latest|build:\n      context: ./server|; \     # Modify services/server to build locally instead of pulling from DockerHub
+             s|image: szurubooru/client:latest|build:\n      context: ./client|; \     # Modify services/client to build locally instead of pulling from DockerHub
+             s|depends_on:\n      - sql|depends_on:\n      - sql\n      - elasticsearch|; \ # Add dependency for alternate arm compatible elasticsearch defined below.
+             s|POSTGRES_HOST: sql|POSTGRES_HOST: sql\n      ESEARCH_HOST: elasticsearch|" \ # Point Postgres to the alternate elasticsearch container
+            docker-compose.yml
+    # "Add a new section to define the alternate elasticsearch service
```

Comment:
You can drop this part, we haven't used elasticsearch in quite a while

Reply:
Removed. :)